parsel/selector.py

```diff
@@ -223,6 +237,44 @@ def __getstate__(self):
     def _get_root(self, text, base_url=None):
         return create_root_node(text, self._parser, base_url=base_url)
 
+    def jmespath(self, query, **kwargs):
+        """
+        Call the ``.jmespath()`` method for each element in this list and return
+        their results flattened as another :class:`SelectorList`.
+
+        ``query`` is the same argument as the one in :meth:`Selector.jmespath`
+
+        Requires : jmespath(https://github.com/jmespath/jmespath)
+        """
+        try:
+            if self.json is not None:
+                datas = self.json
+            elif self._text is not None:
+                datas = json.loads(self._text)
+            else:
+                if isinstance(self.root, six.string_types):
+                    datas = json.loads(self.root)
+                else:
+                    datas = json.loads(self.root.text)
+        except jmespath.exceptions.JMESPathError as exc:
+            msg = u"jmespath error: %s in %s" % (exc, query)
+            msg = msg if six.PY3 else msg.encode('unicode_escape')
+            six.reraise(ValueError, ValueError(msg), sys.exc_info()[2])
+        else:
+            result = jmespath.search(query, datas, **kwargs)
+
+        if type(result) is not list:
+            result = [result]
+
+        def make_selector(x):  # closure function
+            if isinstance(x, six.text_type):
+                return self.__class__(text=x, json=x, _expr=query, type=self.type)
+            else:
+                return self.__class__(json=x, _expr=query, type=self.type)
+
+        result = [make_selector(x) for x in result if x is not None]
```

Comment:
Why filter out `None` values? I think this may be an issue when querying for JSON data that may include `null` values. If the goal is not to return data when search returns `None`, it may be better to add an `if result is None: result = []` earlier.